.github/workflows/smoke-test-release.yml

```diff
@@ -1,185 +1,690 @@
 name: Smoke test release
 on:
+    release:
+        types: [published]
     workflow_dispatch:
         inputs:
-            release_id:
-                description: 'WooCommerce Release Id'
+            tag:
+                description: 'WooCommerce Release Tag'
                 required: true
-
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
 permissions: {}
+env:
+    E2E_WP_LATEST_ARTIFACT: e2e-wp-latest--run-${{ github.run_number }}
+    E2E_UPDATE_WC_ARTIFACT: e2e-update-wc--run-${{ github.run_number }}
+    FORCE_COLOR: 1
 
 jobs:
-    login-run:
-        name: Daily smoke test on release.
+    get-tag:
+        name: Get WooCommerce release tag
```

Comment:
The very first job is about verifying the release version number it received from either `workflow_dispatch` or `release` events, but most especially from `workflow_dispatch` because it's the most prone to invalid manual inputs.